recipes/sml/all/conanfile.py

```diff
@@ -13,37 +13,43 @@ class SMLConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     settings = "compiler"
     no_copy_source = True
+    exports_sources = ['patches/*']
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
-    @property 
-    def _minimum_compilers_version(self): 
+    @property
+    def _minimum_compilers_version(self):
         return {
-            "Visual Studio": "15", 
-            "gcc": "5", 
-            "clang": "5", 
-            "apple-clang": "5.1", 
-        } 
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
 
     def configure(self):
-        if self.settings.compiler.cppstd: 
+        if self.settings.compiler.cppstd:
             check_min_cppstd(self, "14")
-        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False) 
-        if not minimum_version: 
-            self.output.warn("SML requires C++14. Your compiler is unknown. Assuming it supports C++14.") 
-        elif tools.Version(self.settings.compiler.version) < minimum_version: 
-            raise ConanInvalidConfiguration("SML requires C++14, which your compiler does not support.") 
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn(
+                "SML requires C++14. Your compiler is unknown. Assuming it supports C++14.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "SML requires C++14, which your compiler does not support.")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
+        tools.patch(patch_file="patches/clang12.patch")
```

Comment:
This needs to be moved to build like 

https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/libmorton/all/conanfile.py#L27-L29

Reply:
I do not know where to put the for patches. There is no build function. How to deal with patches if its header only?

Reply:
Is the idea to change the source in the build step?